pkg/util/ec2/ec2.go

```diff
@@ -246,8 +268,9 @@ func getToken() (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("unable to read response body, %s", err)
 	}
-
-	return string(all), nil
+	token.value = string(all)
+	token.expirationDate = time.Now().Add(tokenLifetime)
```

Comment:
Super tiny nit: since this is happening after the token is generated on the backend, and after the request is complete, etc... and that might all take a little bit of time, I think it's safer if we calculate the new `token.expirationDate` before the request just to make sure we don't go _over_ the deadline, otherwise the logic at the top of `getToken()` might say we have a valid token when in fact it has already expired on the EC2 side.

Reply:
Good point, I will relocate the expiration date refresh before issuing the request.